libavformat/dashdec.c

```diff
@@ -1725,18 +1782,27 @@ static int get_current_fragment(struct representation *pls, struct fragment** ne
     if (c->is_live) {
         min_seq_no = calc_min_seg_no(pls->parent, pls);
         max_seq_no = calc_max_seg_no(pls, c);
+        old_max_seq_no = max_seq_no;
 
         if (pls->timelines || pls->fragments) {
             err = refresh_manifest(pls->parent);
             if (AVERROR_INPUT_CHANGED == err) {
                 return err;
             }
+            min_seq_no = calc_min_seg_no(pls->parent, pls);
+            max_seq_no = calc_max_seg_no(pls, c);
         }
         if (pls->cur_seq_no < min_seq_no) {
-            av_log(pls->parent, AV_LOG_VERBOSE, "old fragment: cur[%"PRId64"] min[%"PRId64"] max[%"PRId64"]\n", (int64_t)pls->cur_seq_no, min_seq_no, max_seq_no);
+            av_log(pls->parent, AV_LOG_VERBOSE, "old fragment: cur[%"PRId64"] min[%"PRId64"] max[%"PRId64"]\n",
+                   (int64_t)pls->cur_seq_no, min_seq_no, max_seq_no);
             pls->cur_seq_no = calc_cur_seg_no(pls->parent, pls);
-        } else if (pls->cur_seq_no > max_seq_no) {
-            av_log(pls->parent, AV_LOG_VERBOSE, "new fragment: min[%"PRId64"] max[%"PRId64"]\n", min_seq_no, max_seq_no);
+        } else if (pls->cur_seq_no > old_max_seq_no) {
+            av_log(pls->parent, AV_LOG_VERBOSE, "new fragment: cur[%"PRId64"] min[%"PRId64"] max[%"PRId64"] (prev_max[%"PRId64"])\n",
+                   (int64_t)pls->cur_seq_no, min_seq_no, max_seq_no, old_max_seq_no);
+            if (pls->cur_seq_no > max_seq_no) {
```

Comment:
Is this the live edge case, where we're still waiting for the segment to be available?